### PR TITLE
Updates to Ch13 - Readability, wording, grammar

### PR DESCRIPTION
**Pull request by Xmasreturns**
Renamed 13.7.1 "The problem" -> "The Problem"
Removed some unhelpful code and reworded some description to be less redundant between sections

### CODE REVIEW

File: examples/generics/assoc_items/input.md [hunk 1]
```diff
@@ -1,19 +1,13 @@
...
 
 ### See also:
 
-[RFC](
-https://github.com/rust-lang/rfcs/blob/master/text/0195-associated-items.md
-)
+[RFC][RFC]
+
 [items]: http://doc.rust-lang.org/reference.html#items
+[RFC]: https://github.com/rust-lang/rfcs/blob/master/text/0195-associated-items.md
```
Comment (mdinger) [inline examples/generics/assoc_items/input.md]: I think this was a carryover from when the max line lengths were shorter and the parser wouldn't accept `[abc]:` and `url` on different lines (as a historical note).

Comment (Xmasreturns) [inline examples/generics/assoc_items/input.md]: Any idea if this is still the case? If so, then I can revert this to make sure it passes checks.

**Comment by mdinger** [comment]
This is really really great!

File: examples/generics/assoc_items/input.md (hunk 1 above)
Comment (mdinger) [inline examples/generics/assoc_items/input.md]: If you're building the docs, then just manually check the page to make sure the link appears properly. If it does, it should be fine. If you aren't building the docs, make sure the line length is less than [100 characters](https://github.com/rust-lang/rust-by-example/blob/master/check-line-length.sh#L8) (actually, I'm not sure that's the right link but I couldn't find any others). Maybe we don't check md files for line length...I thought we did...

Comment (Xmasreturns) [inline examples/generics/assoc_items/input.md]: Line should be 83 characters, so I think we're good here

**Comment by steveklabnik** [comment]
:confetti_ball:

**Pull request merged**